app/controllers/api/v1/subjects_controller.rb

```diff
@@ -108,15 +109,22 @@ def location_params(locations)
     end
   end
 
-  def selector_context
+  def selector_context(selected_subject_ids)
     if Panoptes.flipper[:skip_subject_selection_context].enabled?
       {}
     else
+      if api_user.logged_in?
```

Comment:
This would be great if it was just `FavoritesFinder.new(api_user, project).ids` and that handled the internal logic.